sunkit_spex/extern/rhessi.py

```diff
@@ -203,15 +205,18 @@ def _update_srm_state(self):
                     "covers attenuator state change. Don't trust this fit!"
                 )
 
-        new_att_state = 0
-        for i in range(len(self._attenuator_state_info['states']) - 1):
-            state = self._attenuator_state_info['states'][i]
-            if change_times[i] < start_time and end_time < change_times[i+1]:
-                new_att_state = state
-                break
-
+        n_states = len(self._attenuator_state_info['states'])
+        new_att_state = self._attenuator_state_info['states'][0]  # default to first
+        if n_states > 1:
+            for i in range(n_states-1):
+                state = self._attenuator_state_info['states'][i]
+                if change_times[i] < start_time and end_time < change_times[i+1]:
+                    new_att_state = state
+                    break
```

Comment:
So when trying to use the older RHESSI example this broke this now work with both but not sure how robust it is.